core/trino-main/src/main/java/io/trino/testing/assertions/TrinoExceptionAssert.java

```diff
@@ -84,6 +88,27 @@ private static boolean isTrinoException(String type)
         return isTrinoExceptionCache.getUnchecked(type);
     }
 
+    private static Optional<FailureInfo> getFailureInfo(Throwable throwable)
+    {
+        return switch (throwable) {
+            case TrinoException trinoException -> Optional.of(toFailure(trinoException).toFailureInfo());
+
+            case QueryFailedException queryFailedException -> {
+                if (queryFailedException.getCause() == null) {
+                    yield Optional.empty();
+                }
+                Optional<FailureInfo> failureInfo = switch (queryFailedException.getCause()) {
```

Comment:
you can yield a switch here too:

```
			yield switch (queryFailedException.getCause()) {
                    case Failure failure -> Optional.of(failure.getFailureInfo().toFailureInfo());
                    case FailureException failure -> Optional.of(failure.getFailureInfo());
                    default -> Optional.empty();
                };
```
ain't pretty but you assign to a variable just to yield it right away

Reply:
agreed and i had that. but this confuses checkstyle indentation rules. i would had to misformat the code to make checkstyle happy. a battle for another time